internal/cmd/providers.go

```diff
@@ -35,24 +35,68 @@ func newProvidersCmd(cli *CLI) *cobra.Command {
 	return cmd
 }
 
+type providerAPIOptions struct {
+	PrivateKey  string
+	ClientEmail string
+	DomainAdmin string
+}
+
+func (o providerAPIOptions) Validate(providerKind string) error {
+	if providerKind != "google" {
+		// Parameters to configure API calls are currently only applicable to Google
+		inapplicableFields := []string{}
+		if o.ClientEmail != "" {
+			inapplicableFields = append(inapplicableFields, "clientEmail")
+		}
+		if o.DomainAdmin != "" {
+			inapplicableFields = append(inapplicableFields, "domainAdmin")
+		}
+		if o.PrivateKey != "" {
+			inapplicableFields = append(inapplicableFields, "privateKey")
+		}
+
+		if len(inapplicableFields) > 0 {
+			return fmt.Errorf("field(s) %q are only applicable to Google identity providers", inapplicableFields)
+		}
+	}
+	return nil
+}
+
+type providerEditOptions struct {
+	ClientSecret       string
+	ProviderAPIOptions providerAPIOptions
+}
+
+func (o providerEditOptions) Validate(providerKind string) error {
+	if o.ClientSecret == "" && o.ProviderAPIOptions.PrivateKey == "" && o.ProviderAPIOptions.ClientEmail == "" && o.ProviderAPIOptions.DomainAdmin == "" {
+		return fmt.Errorf("Please specify a field to update.'\n\n%s", newProvidersEditCmd(nil).UsageString())
+	}
+
+	if providerKind != "google" && o.ClientSecret == "" {
+		return fmt.Errorf("Client secret flag must be specified when updating an identity provider that isn't of kind Google")
+	}
+
+	return o.ProviderAPIOptions.Validate(providerKind)
+}
+
 func newProvidersEditCmd(cli *CLI) *cobra.Command {
-	var secret string
+	var opts providerEditOptions
 
 	cmd := &cobra.Command{
 		Use:   "edit PROVIDER",
 		Short: "Update a provider",
 		Example: `# Set a new client secret for a connected provider
-$ infra providers edit okta --client-secret`,
+$ infra providers edit okta --client-secret VT_oXtkEDaT7UFY-C3DSRWYb00qyKZ1K1VCq7YzN`,
 		Args: ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if secret == "" {
-				return fmt.Errorf("Please specify a field to update.'\n\n%s", cmd.UsageString())
-			}
-			return updateProvider(cli, args[0], secret)
+			return updateProvider(cli, args[0], opts)
 		},
 	}
 
-	cmd.Flags().StringVar(&secret, "client-secret", "", "Set a new client secret")
+	cmd.Flags().StringVar(&opts.ClientSecret, "client-secret", "", "Set a new client secret")
+	cmd.Flags().StringVar(&opts.ProviderAPIOptions.PrivateKey, "private-key", "", "The private key used to make authenticated requests to Google's API")
```

Comment:
I would recommend making this `--service-account--key` and point to a file, then loading the private key from that.

Reply:
`types.StringOrFile` might be good here. That lets the user specify the value directly (maybe from an env var), but also allows them to specify a filename that contains the value.

Reply:
Updated to `StringOrFile`, needed to add some extra json parsing to it